### PR TITLE
Extract layout manager to ILayoutRoot

### PR DESCRIPTION
**Pull request by kekekeks**
Part of #1013

### CODE REVIEW

**Comment by grokys** [comment]
@kekekeks i merged master and fixed the failing tests. Is there anything else you wanted to do on this PR?

**Comment by jkoritzinsky** [comment]
What's the status on this PR?

**Comment by nc4rrillo** [comment]
This LGTM and is a very straight forward change.

**Review by jkoritzinsky — APPROVED** [review]

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -147,6 +148,18 @@ public Size ClientSize
             protected set { SetAndRaise(ClientSizeProperty, ref _clientSize, value); }
         }
 
+        protected virtual ILayoutManager CreateLayoutManager() => new LayoutManager();
```
Comment (grokys) [inline src/Avalonia.Controls/TopLevel.cs]: Could we move this down with the other methods? I don't like methods in the middle of properties!

**Review by grokys — CHANGES_REQUESTED** [review]
Just a small nit!

File: src/Avalonia.Controls/TopLevel.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/Avalonia.Controls/TopLevel.cs]: Fixed!

**Pull request merged**